### PR TITLE
Add sirocco package to conda

### PR DESCRIPTION
**Pull request by enriqueartal**
This is my first try, there are probably plenty of errors.!
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [X] Source is from official source.
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [X] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/sirocco/build.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+autoreconf --install
+./configure --prefix=$PREFIX --libdir=$PREFIX/lib
```
Comment (isuruf) [inline recipes/sirocco/build.sh:4]:
```suggestion
./configure --prefix=$PREFIX --libdir=$PREFIX/lib --disable-static
```

File: recipes/sirocco/build.sh [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+make -j${CPU_COUNT}
+make install
```
Comment (isuruf) [inline recipes/sirocco/build.sh:6]: Add `make check` after this?

Comment (enriqueartal) [reply recipes/sirocco/build.sh:6]: This is my first conda package (and not really mine) and I followed what you did for libbraiding; I agree with all the changes

File: recipes/sirocco/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/miguelmarco/SIROCCO2/releases/download/{{ version }}/libsirocco-{{ version }}.tar.gz
+  sha256: e86c9fe3b1f2cf0d3a7381d21c1fc4670f14b538c60c7cc6410ccb0f5b1fbaf8
+
+build:
+  number: 0
```
Comment (isuruf) [inline recipes/sirocco/meta.yaml:13]:
```suggestion
  number: 0
  skip: True  # [win]
```

**Pull request merged**